### PR TITLE
Linux desktop

### PR DESCRIPTION
**Pull request by francisco-dlp**
Resubmission of #135

### CODE REVIEW

**Comment by francisco-dlp** [comment]
In principle it's done. It works for Linux. Can somebody test it in Windows?

**Comment by ericpre** [comment]
I can try it on windows. Could you add some documentation?

It would be to have it available from a menu ("Settings" menu maybe?) but we can leave it for another PR.

**Comment by francisco-dlp** [comment]
Documentation updated. 

I don't know if it is worth to make this available through the menu. Enabling and disabling the integration can be done automatically when using conda and the bundle and ``pip`` users aren't typically intimidated by the command line.

File: hyperspyui/desktop_integration.py [hunk 1]
```diff
@@ -1,175 +1,22 @@
...
-# along with HyperSpyUI.  If not, see <http://www.gnu.org/licenses/>.
-"""
-Created on Sun Apr 12 17:43:47 2015
-
-@author: Vidar Tonaas Fauske
-"""
-
 import sys
-import os
-import platform
```
Comment (ericpre) [inline hyperspyui/desktop_integration.py]: This import is used for windows.

**Comment by ericpre** [comment]
It seems that it doesn't work completely on windows... but it looks to me (not easy to be sure from the diff) that you just moved the windows part from on file to another, if so it should not be related to this PR.

**Comment by vidartf** [comment]
@ericpre Which bits does not work? The file-ext registration, the shortcut creation, or both?

**Comment by ericpre** [comment]
Anaconda (win7 and win10):
- the shortcut in the start menu: it is created but not working and one reason may be necessary to activate the environment before
- the file extension registration possibly work but it is not sure because of the previous issue (the icon of the file changed, though)
- the remove option (shortcut and file extension) is not working but this was not implemented previously

Winpython
- the shortcuts work
- the file extension registration doesn't seem to work

**Comment by ericpre** [comment]
For the desktop integration in conda, we could the same approach as in https://github.com/hyperspy/start_jupyter_cm/pull/8. And I would not worry too much about the shortcuts because it may better to use menuinst: https://github.com/ContinuumIO/menuinst.

File: hyperspyui/desktop_integration.py [hunk 2]
```diff
@@ -1,175 +1,22 @@
...
-import subprocess
-
-
 import argparse
 
 parser = argparse.ArgumentParser(
     description='Install HyperspyUI into the OS')
 parser.add_argument('-remove', action="store_true", help='uninstall flag')
 parser.add_argument('-no-shortcuts', action="store_true",
```
Comment (ericpre) [inline hyperspyui/desktop_integration.py]: Not really sure about this, but I think that `--` is more usual than `-` (which is usually for single letter argument)?

Comment (francisco-dlp) [reply hyperspyui/desktop_integration.py]: I've seen both styles in BASH commands but I agree that ``-`` is most often reserved to single letter args. I'll change it.

**Comment by francisco-dlp** [comment]
I think that this PR does what it advertises, should we merge it and fix the Win integration issues in a separate pull request?

File: doc/source/installation.rst [hunk 3]
```diff
@@ -40,7 +29,7 @@ run:
 Append the ``--help`` flag for other options.
 
 Currently, this only
-integrates with Windows, but this is intended to be extend to Linux systems
+integrates with Windows and Linx, but this is intended to be extend to MacOS systems
```
Comment (ericpre) [inline doc/source/installation.rst]:
```suggestion
integrates with Windows and Linux, but this is intended to be extend to MacOS systems
```

**Comment by ericpre** [comment]
> I think that this PR does what it advertises, should we merge it and fix the Win integration issues in a separate pull request?

Yes, I agree.

@vidartf: i am not a big fan of the windows registry, so if you could have a quick look/test yourself the extension registration, this would be appreciated! :)

**Pull request merged**